Orange/widgets/visualize/utils/widget.py

```diff
@@ -490,10 +490,10 @@ def setup_plot(self):
 
     # Selection
     def apply_selection(self):
-        if self.data is not None and self.__pending_selection is not None \
-                and self.graph.n_valid:
-            index_group = [(index, group) for index, group in
-                           self.__pending_selection if index < len(self.data)]
+        pending = self.__pending_selection
+        if self.data is not None and pending is not None and len(pending) \
+                and max(i for i, _ in pending) < self.graph.n_valid:
+            index_group = [(index, group) for index, group in pending]
```

Comment:
What's the effect of this line? If it's needed (is it?), isn't it the same as `index_group = list(pending)`?